tubular/comparison.py

```diff
@@ -1,7 +1,12 @@
-import pandas as pd
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
 
 from tubular.base import BaseTransformer
 
+if TYPE_CHECKING:
```

Comment:
This will have been to appease the `TCH001` [typing-only-first-party-import lint](https://docs.astral.sh/ruff/rules/typing-only-first-party-import/) as part of the TCH set.

The pandas API isn't directly used at runtime so, for the code in just this sub-class, pandas is not needed. Hence the linter is asking for it to be placed in a special block that only the type hinter sees. But its parent need pandas, so it's a bit of a moot point.

We should probably:
i.) Use the pandas API directly (perhaps [pandas.Series.equals](https://pandas.pydata.org/docs/reference/api/pandas.Series.equals.html#pandas-series-equals)?) rather than `==`. Out of scope and a ticket in its own right, alongside a test showing where `==` is problematic
ii.) Keep pandas as an import (it's core to the package) and just enable `# noqa: TCH001` next to it.

Reply:
Elected to use option ii. until such time that we enable option i.

An arguably superior way to exempt classes from being moved into a `TYPE_CHECKING` block would be to use the [runtime-evaluated-base-classes](https://docs.astral.sh/ruff/settings/#flake8-type-checking-runtime-evaluated-base-classes) option in `ruff.toml`, but that would require a higher version of `ruff` than the project currently uses. Not for this PR.